src/__experimental__/components/checkbox/checkbox.spec.js

```diff
@@ -327,4 +328,19 @@ describe('Checkbox', () => {
       });
     });
   });
+  it('the required prop is passed to the input', () => {
+    const wrapper = mount(
+      <Checkbox
+        value='foo'
+        label='Required'
+        required
+      />
+    );
+
+    const input = wrapper.find('input');
+    expect(input.prop('required')).toBe(true);
+
+    const label = wrapper.find(Label);
+    expect(label.prop('isRequired')).toBe(true);
+  });
```

Comment:
I think that should be split into two tests, one for the input and one for the label

Reply:
I think having it in two tests is unnecessary because the setup code is the same.
It's asserting that one thing (`required` prop) does two things. I'll update the test description to better reflect that.

Reply:
I disagree, as you are passing that prop to two separate components, it does not say which one failed based on test output. As for the setup code, it could be set in one block:

```suggestion
  describe(`when the require prop is provided`, () => {
    let wrapper;
    
    beforeEach(() => {
      wrapper = mount(
        <Checkbox
          value='foo'
          label='Required'
          required
        />
      );    
    });
    
  it('the required prop is passed to the input', () => {
    const input = wrapper.find('input');
    expect(input.prop('required')).toBe(true);
  });

  it('the label has the isRequired prop', () => {
    const label = wrapper.find(Label);
    expect(label.prop('isRequired')).toBe(true);
  });
```

Reply:
I think it is pretty clear, but I'll make these changes using a `beforeAll` if you agree? I would like to avoid the setup and tare-down of the component unnecessarily.

```
  ● Textbox › the required prop is passed to the input and the label is required

    expect(received).toBe(expected) // Object.is equality

    Expected: false
    Received: true

      125 | 
      126 |     const label = wrapper.find(Label);
    > 127 |     expect(label.prop('isRequired')).toBe(false);
          |                                      ^
      128 |   });
      129 | 
      130 |   describe('Prefix', () => {

      at Object.toBe (src/__experimental__/components/textbox/textbox.spec.js:127:38)
```

Reply:
I agree, it would be better to use `beforeAll` here.
As for the test I would like to point out that you'll get the report of the first assertion failing, to get to the second one you'll need to fix what causing the failure of the first one as testing will stop there. If we have two separate cases, we'll get two separate failures which we can fix without re-running tests.
When we are separating tests like that it might give us a hint where something could go wrong.

Reply:
All changes made, ready for re-review.